pkgs/development/compilers/chicken/5/overrides.nix

```diff
@@ -84,6 +80,38 @@ in {
   zmq = addToBuildInputs pkgs.zeromq;
   zstd = addToBuildInputs pkgs.zstd;
 
+  # less trivial fixes
+  git = old: (addToBuildInputsWithPkgConfig pkgs.libgit2 old) // {
+    postPatch = ''
+      substituteInPlace libgit2.scm \
+        --replace "asize" "reserved"
+    '';
+  };
+  lazy-ffi = old: (addToBuildInputs pkgs.libffi old) // {
+    postPatch = ''
+      substituteInPlace lazy-ffi.scm \
+        --replace "ffi/ffi.h" "ffi.h"
+    '';
+  };
+  opengl = old:
+    (addToBuildInputsWithPkgConfig
+      ([ pkgs.libGL pkgs.libGLU ]
+      ++ lib.optionals stdenv.isDarwin [ pkgs.darwin.apple_sdk.frameworks.Foundation ])
+      old)
+    // {
+      postPatch = ''
+        substituteInPlace opengl.egg \
+          --replace '"-framework OpenGL"' '"-framework" "OpenGL"' \
+          --replace '"-framework Foundation"' '"-framework" "Foundation"'
+      '';
+    };
```

Comment:
You can do it in one go
```nix
    {
       postPatch = old.postPatch or "" + ''
         substituteInPlace opengl.egg --replace 'framework ' 'framework" "'
         '';
     }
     // (addToBuildInputsWithPkgConfig ([ pkgs.libGL pkgs.libGLU ]
     ++ lib.optionals stdenv.isDarwin (with pkgs.darwin.apple_sdk.frameworks; [ OpenGL Foundation ])) old);
```

https://github.com/Et7f3/nixpkgs/commit/cd258d3bb8741ebcd243b4d26c8accd4962ada98